docs/kubernetes_basic.md

```diff
@@ -132,3 +132,29 @@ my-app-a9f44174fa720b22-exec-3   1/1     Running   0          89s
 
 The moment we close the Clojure REPL session, the worker pods get deleted in Kubernetes.
 
+## Reading files into Spark on Kubernetes
+
+We have a lot of different options for reading files into Spark in a Kubernetes setup.
+The main requirement is, that the files can be addressed under a common URL in the driver and all workers.
+
+In Geni all file reading happens via functions of form  `g/read_xxx_!`. We can use different (eventually remote) file sytems by using different forms of the URL (specifically by using different prefixes).
+
+The reading of  files by Geni can be achieved in 4 different ways:
+
+0. (Just copy the data files into the same path in all the Kubernetes pods and the driver machine. This is very adhoc and likely not suitable in most scenarios, as the data files would be lost at the end of the REPL session due to pods being terminated.)
+ 
+1. (Add data files to custom  Docker images used for Driver and Worker nodes. This is rather inconvenient as well, as any change in the data requires different Docker images)
+ 
+2. Mount remote filesystem into the pod on operating system level. This makes them accessible as local files. Example: [blobfuse](https://github.com/Azure/azure-storage-fuse). This has direct support for Kubenetes as well: [kubernetes volume driver](https://github.com/Azure/kubernetes-volume-drivers) (This is then the same as 3.)
+ 
+3. __Mount Kubernetes data volumes into pods:__ Using a __local__ file path __and__ make sure that these path exist and work in all workers __and__ the driver. In Kubernets/Docker we can do this by __mounting__ (eventually remote) filesystems into the pods at certain location. Doing this, the files would look like local files and urls would not have a specific prefix, and be for example `(g/read "/data/test.csv")` This approach can eventually become easier, if we make sure, that the driver is running as well inside Kubernetes. (In the minikube setup above, the driver runs __outside__ Kubernetes)
```

Comment:
I think it's nice to make the gist of the point bold. Should we do it to points 0, 1 and 2 as well?

"easier, if" -> "easier if"

And also, let's remove the final parens.